### PR TITLE
ensure_unlocked_desktop: handle the case where user is not pre-selected

### PR DESCRIPTION
**Pull request by ggardet**
on gnome

It will fix https://openqa.opensuse.org/tests/761827#step/keymap_or_locale/4 and https://openqa.opensuse.org/tests/761827#step/consoletest_finish/11

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -418,7 +418,20 @@ sub ensure_unlocked_desktop {
                 type_string "$username";
                 save_screenshot;
             }
-            send_key 'ret';
+            # On gnome, user may not be selected and using 'ret' is not enough in this case
+            if (check_var('DESKTOP', 'gnome') && (is_sle('15+') || is_leap('15.0+') || is_tumbleweed)) {
```
Comment (rwx788) [inline lib/utils.pm]: We get same code in `handle_login`, this code is used everywhere, so we have to be careful by changing it too.

Comment (rwx788) [reply lib/utils.pm]: We could at least extract same code to separate method. But I would like to see verification runs for SLE with gnome and textmode and different desktops for any openSUSE distri. This is used in too many places and is pretty fragile to such changes =(

Comment (ggardet) [reply lib/utils.pm]: I do not have access to SLE tests. Could you run the tests for me?

Comment (rwx788) [reply lib/utils.pm]: Sure, I can support you with that, but let's solve point above and get verification runs for openSUSE first.

**Comment by ggardet** [comment]
New patch version is being tested on TW with gnome-ext4 on aarch64.

**Comment by ggardet** [comment]
I tested it successfully on Tumbleweed aarch64. @rwx788 , could you test it on SLE, please?

**Comment by rwx788** [comment]
@ggardet on my list, will update you once I'm done

File: lib/utils.pm [hunk 2]
```diff
@@ -406,6 +407,20 @@ sub workaround_type_encrypted_passphrase {
...
+# Handle the case when user is not selected, on gnome
+sub handle_user_not_selected {
+    my ($myuser) = @_;
+    $myuser //= $username;
+    assert_screen [qw(displaymanager-user-selected displaymanager-user-notselected)];
+    if (match_has_tag('displaymanager-user-notselected')) {
+        assert_and_click "displaymanager-$myuser";
+        record_soft_failure 'bsc#1086425- user account not selected by default, have to use mouse to login';
+    }
+    elsif (match_has_tag('displaymanager-user-selected')) {
```
Comment (asmorodskyi) [inline lib/utils.pm]: I am confused - function name saying that you handle not selected user, but you using needle with name "user-selected"

Comment (asmorodskyi) [reply lib/utils.pm]: I would recommend rename a function to `select_user_gnome`

Comment (ggardet) [reply lib/utils.pm]: @asmorodskyi Done in latest version

**Comment by asmorodskyi** [comment]
LGTM , but waiting prove run from @rwx788

**Comment by rwx788** [comment]
VR: 
[sle 12 gnome](http://g226.suse.de/tests/2643)
[sle 12 minimalx](http://g226.suse.de/tests/2644)
[sle 15 gnome](http://g226.suse.de/tests/2648)
[sle 15 minimalx](http://g226.suse.de/tests/2647)

**Pull request merged**